### PR TITLE
Hide the assessment tab bar when there is only one tab

### PR DESCRIPTION
**Pull request by harryggg**
Fix issue #681

### CODE REVIEW

**Comment by lowjoel** [comment]
I think this logic should be moved to the view helper, and tested with a spec.

File: app/views/course/assessment/assessments/index.html.slim [hunk 1]
```diff
@@ -1,9 +1,9 @@
 = page_header format_inline_text(@category.title) do
   = new_button([current_course, :assessment]) if can?(:create, Course::Assessment.new(tab: @tab))
-
-= tabs do
-  - @category.tabs.each do |item|
-    = nav_to(format_inline_text(item.title),
+- if has_more_than_one_tab?
+  = tabs do
```
Comment (lowjoel) [inline app/views/course/assessment/assessments/index.html.slim]: Maybe I'm not clear enough.

I think the entire construction of tabs should be moved to the helper, so the helper is in charge of rendering the tabs (or returning nothing if there are no tabs)

This allows you to directly check the generated output for tabs using HTML matchers instead.

**Comment by lowjoel** [comment]
@harryggg when you're done fixing all the comments (including those from Hound) please squash everything down to one logical commit.

**Comment by harryggg** [comment]
squashed

File: app/views/course/assessment/assessments/index.html.slim [hunk 2]
```diff
@@ -2,9 +2,8 @@
   = new_button([current_course, :assessment]) if can?(:create, Course::Assessment.new(tab: @tab))
 
 = tabs do
-  - @category.tabs.each do |item|
-    = nav_to(format_inline_text(item.title),
-             course_assessments_path(current_course, category: @category, tab: item))
+  - assessment_tabs.each do |item|
```
Comment (lowjoel) [inline app/views/course/assessment/assessments/index.html.slim]: I was thinking more along the lines of this line just being

```ruby
= display_assessment_tabs
```

then in the helper

```ruby
return nil if @category.tabs == 1
tabs do
  @category.tabs.each...
end
```

The current changes doesn't seem idiomatic Ruby -- you've passed a block to tabs, and assessment_tabs is just an array (which is exactly the same as `@category.tabs`). Just place all the HTML logic in the helper.

File: app/helpers/course/assessment/assessments_helper.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+module Course::Assessment::AssessmentsHelper
+  def assessment_tabs
+    tabs_list = []
+    if @category.tabs.count > 1
+      @category.tabs.each do |item|
+        tabs_list.push(nav_to(format_inline_text(item.title),
+                            course_assessments_path(@course, category: @category, tab: item)))
```
Comment (lowjoel) [inline app/helpers/course/assessment/assessments_helper.rb]: Also please clear up the code lints in your next review.

File: app/views/course/assessment/assessments/index.html.slim [hunk 4]
```diff
@@ -1,10 +1,7 @@
 = page_header format_inline_text(@category.title) do
   = new_button([current_course, :assessment]) if can?(:create, Course::Assessment.new(tab: @tab))
 
-= tabs do
-  - @category.tabs.each do |item|
-    = nav_to(format_inline_text(item.title),
-             course_assessments_path(current_course, category: @category, tab: item))
+=display_assessment_tabs
```
Comment (lowjoel) [inline app/views/course/assessment/assessments/index.html.slim]: there should be a space after the =

File: spec/helpers/course/assessment/assessments_helper_spec.rb [hunk 5]
```diff
@@ -0,0 +1,32 @@
+require 'rails_helper'
+
+RSpec.describe Course::Assessment::AssessmentsHelper do
+  let!(:instance) { create(:instance) }
+  with_tenant(:instance) do
+    let(:course) { create(:course) }
+    let(:category) { create(:course_assessment_category, course: course) }
+
+    describe '# assessment_tabs' do
```
Comment (lowjoel) [inline spec/helpers/course/assessment/assessments_helper_spec.rb]: no space after #

File: spec/helpers/course/assessment/assessments_helper_spec.rb [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+      before { helper.instance_variable_set(:@category, category) }
+      before { helper.instance_variable_set(:@course, course) }
+      subject { helper.display_assessment_tabs }
+      context 'if there is exactly 1 tab' do
```
Comment (lowjoel) [inline spec/helpers/course/assessment/assessments_helper_spec.rb]: when there is exactly

File: spec/helpers/course/assessment/assessments_helper_spec.rb [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+        it 'return an empty list' do
```
Comment (lowjoel) [inline spec/helpers/course/assessment/assessments_helper_spec.rb]: returns

File: spec/helpers/course/assessment/assessments_helper_spec.rb [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+          expect(subject).to be_nil
+        end
+      end
+
+      context 'if there are more than 1 tab' do
```
Comment (lowjoel) [inline spec/helpers/course/assessment/assessments_helper_spec.rb]: when there are

File: spec/helpers/course/assessment/assessments_helper_spec.rb [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+        # add an additional tab
```
Comment (lowjoel) [inline spec/helpers/course/assessment/assessments_helper_spec.rb]: this comment is unnecessary

File: spec/helpers/course/assessment/assessments_helper_spec.rb [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+        let!(:tabs) { create_list(:course_assessment_tab, 1, course: course, category: category) }
```
Comment (lowjoel) [inline spec/helpers/course/assessment/assessments_helper_spec.rb]: you can create another tab here, without using create_list

File: app/helpers/course/assessment/assessments_helper.rb [hunk 11]
```diff
@@ -0,0 +1,11 @@
+module Course::Assessment::AssessmentsHelper
+  def display_assessment_tabs
+    return nil if @category.tabs.count == 1
+    tabs do
+      @category.tabs.each do |item|
+        concat(nav_to(format_inline_text(item.title),
+                      course_assessments_path(@course, category: @category, tab: item)))
```
Comment (lowjoel) [inline app/helpers/course/assessment/assessments_helper.rb]: try using current_course instead of `@course`.

**Comment by lowjoel** [comment]
@harryggg sorry, last batch of changes

File: spec/helpers/course/assessment/assessments_helper_spec.rb (hunk 8 above)
Comment (harryggg) [inline spec/helpers/course/assessment/assessments_helper_spec.rb]: uh.

**Comment by harryggg** [comment]
@lowjoel how did this happen, it didn't touch that part...

**Pull request merged**